src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPoolManager.cs

```diff
@@ -299,6 +319,26 @@ public Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, bool doRe
             return SendAsyncCore(request, proxyUri, doRequestAuth, isProxyConnect:false, cancellationToken);
         }
 
+        public async Task<HttpResponseMessage> SendAsyncMultiProxy(HttpRequestMessage request, bool doRequestAuth, MultiProxy multiProxy, CancellationToken cancellationToken)
+        {
+            HttpRequestException rethrowException = null;
+
+            foreach (Uri proxyUri in multiProxy)
+            {
+                try
+                {
+                    return await SendAsyncCore(request, proxyUri, doRequestAuth, isProxyConnect: false, cancellationToken).ConfigureAwait(false);
+                }
+                catch (HttpRequestException ex) when (ex.AllowRetry != RequestRetryType.NoRetry)
```

Comment:
Doesn't this mean it's going to retry on a different proxy even if the exception says it should retry on the same proxy?

Reply:
I see what you mean. Behavior is intentional but I have a misnomer; I will change the name of `RetryOnSameProxy` to `RetryOnSameOrNextProxy`